### PR TITLE
Order and favorite events view

### PR DESCRIPTION
**Pull request by hugolan**

### CODE REVIEW

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 1]
```diff
@@ -194,5 +199,78 @@ public void onNoteClick(int position) {
...
+                    List<Event> listFavorites = new ArrayList<>();
+                    if(isOnOrderedView){
+                        for (Event e : orderedEvents) {
+                            if (favorites.get(e.getId())) listFavorites.add(e);
+                        }
+                    }else{
+                        for (Event e : events) {
+                            if (favorites.get(e.getId())) listFavorites.add(e);
+                        }
+                    }
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:226]: You can rewrite this as 
```
for (Event e : isOnOrderedView ? orderedEvents : events) { 
     if (favorites.get(e.getId())) listFavorites.add(e);
}
```
Which is less heavy.

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 2]
```diff
@@ -194,5 +199,78 @@ public void onNoteClick(int position) {
...
+                    List<Event> listOrdered = new ArrayList<>();
+                    if(isOnFavoriteView){
+                        for (Event event : favoritesEvents) {
+                            listOrdered.add(new Event(event.getId(), event.getName(), event.getLocationName(), event.getCoordinates(), event.getStartTime(), event.getEndTime(), event.getRecurrence()));
+                        }
+                    }else{
+                        for (Event event : events) {
+                            listOrdered.add(new Event(event.getId(), event.getName(), event.getLocationName(), event.getCoordinates(), event.getStartTime(), event.getEndTime(), event.getRecurrence()));
+                        }
+                    }
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:257]: Same here

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 3]
```diff
@@ -194,5 +199,78 @@ public void onNoteClick(int position) {
...
+                    listOrdered.sort((l, r) -> {
+                        if (l.equals(r)) return 0;
+                        return l.getStartTime().isBefore(r.getStartTime()) ? -1 : 1;
+                    });
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:261]: You can use the compareTo(other) of ZonedDateTime (`return l.compareTo(r)`) or
```
return Long.compare(l.getStartTime().toEpochSecond(), r.getStartTime().toEpochSecond());
```
Which is cleaner

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java (hunk 1 above)
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:226]: yes definitely

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java (hunk 3 above)
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:261]: yep that's better

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 4]
```diff
@@ -82,16 +87,16 @@ public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup c
...
-            public void onClick(View view) {
-                Intent intent = new Intent(root.getContext(), EventCreator.class);
-                startActivity(intent);
-            }
-        });
+        FloatingActionButton fabFavorite = (FloatingActionButton) root.findViewById(R.id.fabFavorite);
+        fabFavorite.setOnClickListener(fabFavorite());
+
+        FloatingActionButton fabOrder = (FloatingActionButton) root.findViewById(R.id.fabOrder);
+        fabOrder.setOnClickListener(fabSortTime());
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:99]: Those casts are useless.

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 5]
```diff
@@ -194,5 +199,66 @@ public void onNoteClick(int position) {
...
+                    });
+                    eventList.setAdapter(new EventViewerAdapter(listOrdered, onNoteListener));
+                    isOnOrderedView = true;
+                    orderedEvents = listOrdered;
+                }else{
+                    if(isOnFavoriteView){
+                        eventList.setAdapter(new EventViewerAdapter(favoritesEvents, onNoteListener));
+                    }else{
+                        eventList.setAdapter(new EventViewerAdapter(events, onNoteListener));
+                    }
```
Comment (Ef55) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:258]: Could be written:
```suggestion

eventList.setAdapter(new EventViewerAdapter(
  isOnFavoriteView ? favoritesEvents : events, 
  onNoteListener
));
```

**Review by Ef55 — CHANGES_REQUESTED** [review]
Code-wise, a few little details and some lint warnings to fix and everything should be good.

Functionality-wise, I have a few comments:

- When testing locally, nothing happens when I click on an event, so it seems to me that there is no way to set an event as favorite;
- When clicking the middle buttons, no event is displayed (dunno what it's supposed to do);
- When the list is empty, nothing is displayed (you should write something like "no events").

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java (hunk 4 above)
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:99]: yep

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java (hunk 5 above)
Comment (hugolan) [reply app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:258]: oh yeah this is simpler

**Comment by hugolan** [comment]
> Code-wise, a few little details and some lint warnings to fix and everything should be good.
> 
> Functionality-wise, I have a few comments:
> 
> * When testing locally, nothing happens when I click on an event, so it seems to me that there is no way to set an event as favorite;
> * When clicking the middle buttons, no event is displayed (dunno what it's supposed to do);
> * When the list is empty, nothing is displayed (you should write something like "no events").

yeah I am going to put the case where there no events
the middle button is supposed to order events by time
it is a long click, but I changed it to just click haven't push it yet

File: app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java [hunk 6]
```diff
@@ -240,25 +247,30 @@ public void onClick(View view) {
...
                 if (!isOnOrderedView) {
                     List<Event> listOrdered = new ArrayList<>();
-                    for (Event e : isOnFavoriteView ? orderedEvents : events) {
-                        if (favorites.get(e.getId())) listOrdered.add(e);
+                    for (Event e : isOnFavoriteView ? favoritesEvents : events) {
+                        listOrdered.add(e);
                     }
                     listOrdered.sort((l, r) -> {
-                        if (l.equals(r)) return 0;
+                        //if (l.equals(r)) return 0;
```
Comment (Flechman) [inline app/src/main/java/com/github/onedirection/navigation/fragment/home/HomeFragment.java:254]: Remove the comment if you don't use it anymore

**Review by Flechman — APPROVED** [review]
The code looks good to me, but you could clean it up even more and refactor the logic in some areas.

**Comment by hugolan** [comment]
> The code looks good to me, but you could clean it up even more and refactor the logic in some areas.

yeah when I will merge it with the event in the db I'll proceed to a refactor to make it cleaner but not for now

**Review by Ef55 — APPROVED** [review]

**Pull request merged**